README.md

```diff
@@ -1 +1,101 @@
+
+At this point, we are developing the Miden node for a centralized operator. Thus, the work does not yet include such components as P2P networking and consensus. These also will be added in the future.
+
+## Architecture
+
+![Architecture diagram](./assets/architecture.png)
+
+The Miden node is made up of 3 main components, which communicate over gRPC: 
+- **store:** stores the current state of the chain.
+- **rpc:** serves client requests such as to synchronize with the latest state of the chain or to submit transactions.
+- **block producer:** accepts transactions from the RPC component, creates blocks containing those transactions, and sends them to the store.
+
+All 3 components can either run in one process, or each component can run in its own process. See the [Running the node](#running-the-node) section for more details.
+
+## Usage
+
+Before running the node, you must first generate the genesis file. 
+
+The `miden-node` executable is used to both generate the genesis file, and running the node.
+
+### Installing the node
+
+To install for production use cases, run
+
+```sh
+# Installs `miden-node` executable
+$ cargo install --path node
+```
+
+Otherwise, if only to try the node out for testing, run
+
+```sh
+$ cargo install --features testing --path node
+```
+
+Currently, the only difference between the 2 is how long the `make-genesis` command will take to run (see next subsection).
+
+If you intend on running the node in different processes, you need to install each component separately:
+
+```sh
+# Installs `miden-node-store` executable
+$ cargo install --path store
+
+# Installs `miden-node-rpc` executable
+$ cargo install --path rpc
+
+# Installs `miden-node-block-producer` executable
+$ cargo install --path block-producer
+```
+
+### Generating the genesis file
+
+The contents of the genesis file are currently hardcoded in Rust, but we intend to make those configurable shortly. The genesis block currently sets up 2 accounts: a faucet account for a `POL` token, as well as a wallet account.
+
+To generate the genesis file, run 
+
+```sh
+$ miden-node make-genesis
+```
+
+This will generate 3 files in the current directory: 
+- `genesis.dat`: the genesis file.
+- `faucet.fsk` and `wallet.fsk`: the public/private keys of the faucet and wallet accounts, respectively.
+
+### Running the node
+
+Each executable will require a configuration file. Each directory containing the executables also contains an example configuration file. For example, `node/miden-node-example.toml` is the example configuration file for running all the components in the same process. Notably, the`store.genesis_filepath` field must point to the `genesis.dat` file that you generated in the previous step.
+
+To run all components in the same process:
+
+```sh
+$ miden-node start -c <path-to-config-file>
+```
```

Comment:
We should either mention logging here, or set the default log level to `INFO` so that the user can tell that the node is running. My preference would be to set the default to `INFO`.

In the future, we can also enable `verbose` mode to print out more info.

Reply:
Done! Set the default to `INFO`, which can be changed with `RUST_LOG` env var (but I don't think we need to mention that here?)